---
string[] color
int32[] size
float32[] posX
float32[] posY
float32[] orientation
